index.dd

```diff
@@ -243,20 +243,17 @@ Modern convenience.
 Modeling power.
 Native efficiency.)
 ----
-#!/usr/bin/env rdmd
-import std.range, std.stdio;
-
 // Compute average line length for stdin
 void main()
 {
-    ulong lines = 0, sumLength = 0;
-    foreach (line; stdin.byLine())
-    {
-        ++lines;
-        sumLength += line.length;
-    }
+    import std.range, std.stdio;
+
+    auto sum = 0.0;
+    auto count = stdin.byLine
+        .tee!(l => sum += l.length).walkLength;
```

Comment:
I don't think that would compile, unless you add a `map!(l => l.length)` between the `tee` and the `sum`.